### DISCUSSION TITLE
How to include GeoServer WMS with credentials

### DISCUSSION THREAD

**Discussion by PentamapFlorian** [rnmapbox/maps#2945] [Q&A]
My included GeoServer WMS RasterLayer was working without problems, until I added authentication/credentials to my GeoServer. Whenever I call my tileUrl https://username:password@my.geoserver.com/... I get following error on Android Devices:

> 'Mapbox error', 'Map load failed: {tile-id={x=8, y=5, z=4}, type=tile, message=Failed to load tile: HTTP status code 401, begin=70787167510, source-id=...}', { level: 'error',
>                                                                                                       tag: 'RCTMGLMapView',
>                                                                                                       message: 'Map load failed: {tile-id={x=8, y=5, z=4}, type=tile, message=Failed to load tile: HTTP status code 401, begin=70787167510, source-id=...}' }

The WMS is showing up without errors on iOS devices. The credentials are correct.

Is there any setting I am missing? Do I have to use something like:

>  transformRequest: (url, resourceType)=> {
>         if(resourceType === 'Source' && url.startsWith('http://myHost')) {
>         return {
>             url: url.replace('http', 'https'),
>             headers: { 'my-custom-header': true},
>             credentials: 'include'  // Include cookies for cross-origin requests
>         }
>     }

Help is kindly apreciated.

Best regards,
Florian

**Comment by mfazekas** [comment]
See also #2957
